lfsapi/ntlm.go

```diff
@@ -29,32 +28,42 @@ func (c *Client) doWithNTLM(req *http.Request, credHelper creds.CredentialHelper
 		return res, nil
 	}
 
-	return c.ntlmReAuth(req, credHelper, creds, true)
+	return c.ntlmReAuth(req, credWrapper, true)
 }
 
 // If the status is 401 then we need to re-authenticate
-func (c *Client) ntlmReAuth(req *http.Request, credHelper creds.CredentialHelper, creds creds.Creds, retry bool) (*http.Response, error) {
-	ntmlCreds, err := ntlmGetCredentials(creds)
-	if err != nil {
-		return nil, err
-	}
-
-	res, err := c.ntlmAuthenticateRequest(req, ntmlCreds)
+func (c *Client) ntlmReAuth(req *http.Request, credWrapper creds.CredentialHelperWrapper, retry bool) (*http.Response, error) {
+	// Try SSPI first.
+	res, err := c.ntlmAuthenticateRequest(req, nil)
 	if err != nil && !errors.IsAuthError(err) {
 		return res, err
 	}
 
-	switch res.StatusCode {
-	case 401:
-		credHelper.Reject(creds)
-		if retry {
-			return c.ntlmReAuth(req, credHelper, creds, false)
+	// If SSPI failed, then we need to try the normal.
+	if res.StatusCode >= 300 || res.StatusCode < 199 {
```

Comment:
It might be tidier to reverse this condition and `return res, nil` on success. Then you don't need to indent this block.